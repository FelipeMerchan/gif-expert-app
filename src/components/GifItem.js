import React from 'react';

export const GifItem = ({ title, url }) => {
  return (
    <div>
      <img
        src={url}
        alt={title}
        title={title}
      />
      <p>{title}</p>
    </div>
  )
}
