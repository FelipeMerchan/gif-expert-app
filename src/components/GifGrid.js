import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';

import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <section aria-label="Gif list">
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">Cargando...</p>}

      <div className="card-grid">
        {
          images.map(img => (
            <GifItem
              key={img.id}
              {...img}
            />
          ))
        }
      </div>
    </section>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
