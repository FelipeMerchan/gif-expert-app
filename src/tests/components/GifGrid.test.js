import { shallow } from 'enzyme';
/* @testing-library/jest-dom viene importado por defecto,
  por lo cual no es necesario importarlo. Sin embargo, se importa
  para que VSCode nos ayude con el auto completado de los metodos y propiedades */
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
/* jest.mock finge la llamada a un archivo y controla la informacion que eso va
  a responder */
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {
  const category = 'One Punch'
  test('Debe mostrarse correctamente', () => {
    /* mockReturnValue: cuando se llame la funcion useFetchGifs dentro del
      componente <GifGrid /> va a retornar el valor que le pasemos como parametro a mockReturnValue  */
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
    /* Un mock sirve para fingir. Por ejemplo, que el componente tiene informacion
      que trae de una API.
    */
    const gifs = [
      {
        id: 'ABC',
        title: 'Cualquier cosa',
        url: 'https://localhost/cualqueir/cosa.jpg',
      },
      {
        id: '123',
        title: 'Cualquier cosa',
        url: 'https://localhost/cualqueir/cosa.jpg',
      }
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    /* Verifica si el parrafo que dice loading existe: */
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    /* Al metodo find podemos pasarle el nombre de un componente ('GifItem') */
    expect(wrapper.find('GifItem').length).toBe(gifs.length);
  });
});