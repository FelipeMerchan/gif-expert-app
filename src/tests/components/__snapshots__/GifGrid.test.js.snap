// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<section
  aria-label="Gif list"
>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualqueir/cosa.jpg"
    />
    <GifItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualqueir/cosa.jpg"
    />
  </div>
</section>
`;

exports[`Pruebas en el componente <GifGrid /> Debe mostrarse correctamente 1`] = `
<section
  aria-label="Gif list"
>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</section>
`;
